### PR TITLE
Shrink ViewPager on Map drag

### PR DESCRIPTION
**Pull request by ambardhesi**

### CODE REVIEW

File: app/src/main/java/com/compscieddy/meetinthemiddle/Util.java [hunk 1]
```diff
@@ -40,17 +40,9 @@ public static Bitmap getCroppedBitmap(Context context, Bitmap bitmap) {
     return output;
   }
 
-  public static void rotateFabForward(View view) {
+  public static void rotateView(View view, float angle){
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/Util.java]: more descriptive, like "rotateExpandSectionView" - long names are better than generic names.
@ambardhesi also maybe this should just be a boolean "boolean isOpen" - I don't foresee us ever needing a specific angle

File: app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import android.view.MotionEvent;
+import android.widget.FrameLayout;
+
+/**
+ * Created by ambar on 6/23/16.
+ */
+public  class TouchableWrapper extends FrameLayout {
+
+  private long lastTouched = 0;
+  private static final long SCROLL_TIME = 100L; // 100 Milliseconds
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java]: I'd rename to MIN_SCROLL_THRESHOLD_MILLIS

File: app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        break;
+      case MotionEvent.ACTION_UP:
+        final long now = SystemClock.uptimeMillis();
+        if (now - lastTouched > SCROLL_TIME) {
+          // Update the map
+          userMapDrag.onMapDrag();
+        }
+        break;
+    }
+    return super.dispatchTouchEvent(ev);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java:41]: This still works, I imagine, but we're trying to pass all touch events. Returning true/false determines this, so to be more explicit that we want to pass all touch events to other container views or views beneath, let's return either true or false. Test and see which boolean value works

File: app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    } catch (ClassCastException e) {
+      throw new ClassCastException(context.toString() + " must implement UpdateMapAfterUserInterection");
+    }
+  }
+
+  @Override
+  public boolean dispatchTouchEvent(MotionEvent ev) {
+    switch (ev.getAction()) {
+      case MotionEvent.ACTION_DOWN:
+        lastTouched = SystemClock.uptimeMillis();
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java]: lastTouchedMillis

File: app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        break;
+      case MotionEvent.ACTION_UP:
+        final long now = SystemClock.uptimeMillis();
+        if (now - lastTouched > SCROLL_TIME) {
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/TouchableWrapper.java]: nowMillis (does this have to be a `final` value?)

File: app/src/main/java/com/compscieddy/meetinthemiddle/GroupActivity.java [hunk 6]
```diff
@@ -498,6 +500,39 @@ public void onMapClick(LatLng point) {
...
   @Override
+  public void onMapDrag() {
+    if (!collapsed){
+      Display display = getWindowManager().getDefaultDisplay();
+      Point size = new Point();
+      display.getSize(size);
+      int height = size.y;
+
+      ResizeAnimation resizeAnimation;
+
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/GroupActivity.java]: Don't copy & paste code! Please refactor out this code to make it okay with the code below

File: app/src/main/java/com/compscieddy/meetinthemiddle/Util.java (hunk 1 above)
Comment (ambardhesi) [inline app/src/main/java/com/compscieddy/meetinthemiddle/Util.java]: @eddieberklee I thought this function could be used for rotating any views in the future, not just the FAB, which is why I renamed it to rotateView.
Also, I'm using this function with different angles in the code, which is why the angle parameter is needed.

File: app/src/main/java/com/compscieddy/meetinthemiddle/GroupActivity.java (hunk 6 above)
Comment (ambardhesi) [inline app/src/main/java/com/compscieddy/meetinthemiddle/GroupActivity.java]: @eddieberklee Huh? What do you mean..I need this code.

Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/GroupActivity.java]: @ambardhesi but there's also this code below - move it out into a method so both places just call that method

File: app/src/main/java/com/compscieddy/meetinthemiddle/Util.java (hunk 1 above)
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/Util.java]: @ambardhesi sure ok - that's fine then - to make it clear that's what you wanted to do then (make it more static and more of a helper), then could you move it to Utils

**Pull request merged**